### ISSUE TITLE
java.lang.NoSuchMethodException: crazypants.enderio.base.EnderIO.checkModLists

### ISSUE THREAD

**Issue by iamtpage** [SleepyTrousers/EnderIO#4747]
#### Issue Description:
Note: If this bug occurs in a modpack, please report this to the modpack author. Otherwise, delete this line and add your description here. If you are using an unreleased version of Ender IO (e.g. for 1.8.9), please don't open an issue for features that have not been announced as "finished" or "working".

#### What happens:
Stack trace on server start:
https://gist.github.com/iamtpage/c16360d906b84d0af94203a9e50b74be

#### What you expected to happen:
No stack trace

#### Steps to reproduce:

1. Boot up server
2. Watch server log
3.
...

____
#### Affected Versions (Do *not* use "latest"):

- EnderIO: EnderIO-1.12.2-5.0.29
- EnderCore: EnderCore-1.12.2-0.5.35
- Minecraft: 1.12.2
- Forge: 1.12.2 2705, and SpongeForge (spongeforge-1.12.2-2705-7.1.0-BETA-3260)

#### Your most recent log file where the issue was present: 
https://gist.github.com/iamtpage/9f143d33af8faeccc6b0e407291d9244

**Comment by EpicSquid** [comment]
Sounds like sponge forge to me. Try removing that first. Does this cause a crash or any errors in game?

**Comment by LemADEC** [comment]
Where do we get on fixing this EnderIO issue?

**Comment by HenryLoenwind** [comment]
We have to ways of fixing it:

1) Make it so that Ender IO works in multiplayer if it's only installed on the client and not on the server
2) Get the Forge people to see that solution 1 would be idiotic

**Comment by mindforger** [comment]
can somebod shed a bit light on this? is he using EIO client side without being on the server? shouldn't this be blocked by the handshake?

**Comment by LemADEC** [comment]
@HenryLoenwind we're talking about a server boot issue, before client can even try to connect.
Please elaborate how is that link with client/server synchronization?

**Comment by HenryLoenwind** [comment]
What happens here is that we have code that runs on the client and prevents the "ender io is on the client but not the server" situation. However, that could should not run on the server (obviously...). But Forge sees that we have that code and complains that we don't run it on the server.

This is not a crash, this is a warning by Forge.

(Technical background: A "stack trace" is not a crash. It is just an address. Just the same like "Main Street 4489" is not the location of a fire, unless you call 911 and tell them so. But people who listen to 911-calls all the time may automatically assume so when they hear an address...)

**Comment by mindforger** [comment]
oh wow, okay i missed that part with the warning right at the beginning, so this is not a crash, sorry  to bother then, but i was boggled why it would crash even but it isn't just complaining :)

**Comment by HenryLoenwind** [comment]
np. And if I sound a bit annoyed, then that's carried over from work. We just had a hotfix deployment with a window of acceptable downtime to deploy it of 6am to 8am...and someone put in a database script that ran for over 5 hours...

**Comment by mindforger** [comment]
open doors my friend open doors ... wanna change jobs for a few days? it's just that some co-workers in a remote location change code in drivers without noticing others :D just becasue they needed a quick fix

**Comment by LemADEC** [comment]
ok, I remember now.
Could we remove the `@SideOnly(Side.CLIENT)` annotation and filter the content by the `Side side` parameter instead?

**Comment by HenryLoenwind** [comment]
Yes, but the you would no longer get those nice "Ender IO needs XY at version Z or higher, you only have version F" messages but "Ender IO's network check handler failed".

**Comment by HenryLoenwind** [comment]
As this has been linked in from the Forge ticket, here's an overview about how Forge works at the moment:

`clientSideOnly` --- The mod will not be loaded on servers, it's jar will be injected into the classpath instead.

`serverSideOnly` --- The mod will not be loaded on clients, it's jar will be injected into the classpath instead.

`acceptableRemoteVersions` -- If the mod is on a server, the server will check the connecting clients against this version range and display a nice error message if it doesn't fit. *This key is ignored by the client.*

`@NetworkCheckHandler` -- If it exists, the `acceptableRemoteVersions` key will be ignore. Instead this code is run on both server and client and if it returns `false` a cryptic error message will be displayed.

This gives us the following combinations:

* Client-only mod
* Server-only mod
* Client-only mod with optional server component
* Mod that shows cryptic error messages

**Comment by LemADEC** [comment]
What happens when EnderIO is on the client, but not on a modded server? a vanilla server?

**Comment by tyler489** [comment]
What do you mean what happens?

**Comment by tterrag1098** [comment]
@LemADEC Ideally it would give a nice error explaining that the server is missing EnderIO and that you cannot join it. In the current reality, it gives a generic "NetworkCheckHandler failed" message to *users* which is just poor error handling.

**Comment by tyler489** [comment]
Guis and other things break on server for other mods..

**Comment by LemADEC** [comment]
Mod A is only installed on client, hence its blocks and entities are not available. The only GUI left would be tooltip, HUD and creative menu (?).
Mod B is installed on client and server. Mob A is interacting with its GUI? can we illustrate that with an precise example?

**Comment by tyler489** [comment]
https://github.com/SleepyTrousers/EnderIO/issues/4753
Rs disabled on the server left players unable to open enderios machines.

**Comment by HenryLoenwind** [comment]
In general you can write a mod in a way that nothing bad happens when it's installed client-only. However, if you don't do that (and Ender IO never was written that way) there can be a thousand minor or major issues. I am not willing to go through the whole codebase to first figure out what code could be executed in that case and then find out if that code is client-only-safe.

**Comment by HenryLoenwind** [comment]
### TL;DR: Forge is trying to punish us for doing our version check ourselves. We found a way around that but now Forge is complaining into the logfile. Nothing we can do.
